src/optimagic/exceptions.py

```diff
@@ -47,6 +47,10 @@ class InvalidBoundsError(OptimagicError):
     """Exception for invalid user provided bounds."""
 
 
+class InvalidScalingOptionsError(OptimagicError):
```

Comment:
I would rename this to `InvalidScalingError` so it covers a slightly wider range of errors.